### DISCUSSION TITLE
Getting 'Display Output is not active.' on Mac OS X for Ubuntu 22.04 ARM64 image

### DISCUSSION THREAD

**Discussion by pmessri** [utmapp/UTM#5555] [General]
I'm getting the following:

'Display output is not active.' 

After selecting to install ubuntu server from the latest ubuntu iso image.

In my display setting there is the following selected:  'virtio-gpu-pci'

**Comment by nicolas-g** [comment]
👍 ,  same issue here  ..

**Reply by Graphic-lab** [reply]
Check Mac os X System settings and add UTM.app
- System Settings / Privacy & Security / Accessibility
- System Settings / Privacy & Security / Screen & System Audio Recording

**Reply by jkmaxwell** [reply]
Yes, this worked, thank you.

**Reply by marcusdunlapgit** [reply]
This worked wonderful. Thank you

**Reply by Romain7495** [reply]
It definitely works! Thanks

**Reply by udiNur** [reply]
works for me!! thanks!!

**Reply by shawnz** [reply]
I only needed the second one, "Screen & System Audio Recording". Accessibility permission wasn't necessary for me

**Reply by arwildo** [reply]
Work flawlessly, thank you so much!

**Reply by jfiore-pg** [reply]
Why are these permissions needed?

**Reply by mopoleefans-code** [reply]
My Linux is in showing

**Comment by wade764** [comment]
I am having the same issue as well. Also using the default display adapter 'virtio-gpu-pci'

**Reply by lizishuo28** [reply]
tm艹你妈

**Reply by shoruny** [reply]
??到国外骂街不丢人？

**Reply by startGalway** [reply]
傻逼，丢国人的脸

**Comment by lizishuo28** [comment]
> I'm getting the following:
> 
> 'Display output is not active.' 
> 
> After selecting to install ubuntu server from the latest ubuntu iso image.
> 
> In my display setting there is the following selected:  'virtio-gpu-pci'
> 
> 

纯你妈放屁艹

**Comment by fernandoasqui** [comment]
it worked using virtio-ramfb-gl (GPU Suported)

**Reply by wade764** [reply]
That worked for me as well! Thank you so much.

**Reply by alicebear1203** [reply]
Hi, I had adjusted my setting to virtio-ramfb-gl (GPU Supported), but it stuck.
It now keeps being status in the picture below.
Hope that you can help!
Thanks!
<img width="984" alt="截圖 2023-08-25 下午12 45 53" src="https://github.com/utmapp/UTM/assets/69949230/99997ebb-1778-4202-9e3f-36fc0511fc26">

**Reply by sramac2** [reply]
It worked, thank you so much

**Comment by jbw9964** [comment]
I had similar issue. And it looks like somehow worked.
I used virtio-vga instead.

I tried to emulate ubuntu 16.04.7 LTS (Xenial Xerus) - AMD64, on macOS Ventura (13.5.1) with m1 chip.

The default display setting were virto-gpu-pci, as like others. 
And i tried virtio-ramfb-gl but it stopped (even before booting) more than 5 minutes like @alicebear1203 above.

So I tried to add other display, and the default setting of added display were virtio-vga.
Eventually, I tried virtio-vga instead virtio-ramfb-gl and here's the result.


<img width="600" alt="스크린샷 2023-09-01 15 41 58" src="https://github.com/utmapp/UTM/assets/110011678/c85c4e1a-3ba4-473f-8adc-d273f829cdc4">

<img width="600" alt="스크린샷 2023-09-01 16 04 11" src="https://github.com/utmapp/UTM/assets/110011678/92ad9898-cdfb-4536-9a9a-b930a81de8c1">

---
- full video

https://github.com/utmapp/UTM/assets/110011678/693c70bd-041e-46eb-a02f-76071dee795f

For me, it took about 30 sec to load ubuntu installation (I fasten the vid)

**Reply by alicebear1203** [reply]
Thanks for helping!
I found I am so silly that my goal is to install Ubuntu GUI, so I can just download the GUI iso, and then all done :P

**Reply by SeeTheSeaWithoutFlying** [reply]
Thank you very much. It was very helpful. Thank you

**Reply by arbourd** [reply]
I ran into the same issue with emulating a Kairos amd64 build on a macOS 14.2 device with a M2 processor.

Changing the GPU from `virtio-gpu-gl-pci` to `virtue-vga-gl` worked for me. I am using Hardware acceleration.

**Reply by dimashpt** [reply]
Thank you, it works!

**Reply by losyoply** [reply]
thank you><

**Comment by seungwonme** [comment]
I had encountered the same issue and managed to resolve it by following these steps:

1. I checked the box for `Enable hardware OpenGL acceleration`.
2. Then, I navigated to `Settings` => `Display` => `Emulated Display Card` and selected `virtio-ramfb-gl (GPU Supported)`.

My computer environment is as follows: 

I am using an `M1 Mac with OS Sonoma 14.2.1`, and I encountered this issue while running `Ubuntu ARM 22.04.3` on a `UTM 4.4.5`.

Reference
- [linux - Which Virtual Display for UTM Ubuntu under M1 Mac?](https://superuser.com/questions/1788158/which-virtual-display-for-utm-ubuntu-under-m1-mac)
- [How To Install Debian Linux 11 (BULLSEYE) On ANY M1/M2 Mac Using UTM || Linux On APPLE SILICON](https://www.youtube.com/watch?v=yCTx-Xg8uuQ)

**Reply by arkagme** [reply]
Hey where did you find the option for "Enable Hardware OpenGL Acceleration" ?

**Reply by muya** [reply]
hi @arkagme , I think this option is now labelled ANGLE (OpenGL)

Go to Main app settings -> Display -> Renderer backend

![screenshot-20240712_070357](https://github.com/user-attachments/assets/b8fc9278-c97c-4384-9932-5917638ef337)

**Reply by gaurav9822** [reply]
works for me

**Reply by perfumescent** [reply]
works for win11 thx

**Comment by johndunning** [comment]
I had the same problem but it went when I increased the size of the display window - just drag it with the mouse

**Comment by digevil** [comment]
well I got this message as well but my solution is ... just wait a few seconds(without doing anything) and the expect UI just show up ...

**Reply by lohriialo** [reply]
Same here, just wait : M1, Mac OS 14.6, UTM, 4.5.4

**Comment by EchoEllet** [comment]
Same issue for Fedora KDE (Intel and AMD x86_64), but it's only for a few seconds, waiting a little bit without changing anything like the window dimension or moving the window, and the Fedora boot loading screen will show up but then causing a black screen (stuck):

![image](https://github.com/utmapp/UTM/assets/73608287/fa9dd762-fa35-411a-aeb0-d8de4eef53c4)

**Comment by AleksKash** [comment]
I solved the issue by change display settings on virtio-gpu-vg-pci, bit it takes 2-3 minutes to boot as well every time. Even longer then default set with that display output not active process.

**Comment by FBaxon** [comment]
Is there still someone could help me? I met the same trouble. I didn't mind it yesterday, because it worked after waiting for few seconds when the 'display output is not active' came. But today it didn't run normally as yesterday, just keeping 'display output is not active'. And I tried methods you mentioned, but they didn't work.

the latest UTM was downloaded the official website
to install ubuntu 22.04 arm64
apple m1, macOs sonoma 14.5
at first selected: 'virtio-gpu-pci'

**Reply by FBaxon** [reply]
I finally made it by reinstalling it, and the key was to choose 'GPU acceleration supported' at the beginning

**Comment by Noodlefishh** [comment]
The only thing that worked for me was adding a device type "terminal" in the settings, allowing me to install the OS.

<img width="1242" alt="image" src="https://github.com/user-attachments/assets/61467b86-1d20-4119-8a6d-8333acf59c2d">

**Reply by aditya-lad** [reply]
Thanks, this trick worked for me to proceed with the install for Kali, then I also changed the display to virtio-gpu-gl-pci (GPU Supported) and it works now.

**Reply by hansgilead** [reply]
this is also what I needed.   using home assistant all I could get using any combo of displays was "display output is not active."  adding a new serial device as a built-in terminal allows me to see what's going on and I get the home assistant login prompt now.  thanks!

**Reply by ghost** [reply]
This seems to have been the only solution that worked for me too. Thanks!

**Reply by Anmol-Singh-Jaggi** [reply]
In my case, adding a terminal helped me realize that it was actually waiting for me to enter the disk encryption passphrase.
Thanks!

**Comment by mokagio** [comment]
I installed Ubuntu desktop from the server edition by following [this guide](https://jun1okamura.medium.com/install-ubuntu-on-mac-m1-powered-by-utm-499aba3ba7e9).

After the reboot triggered by `reboot`, the only think I saw in the screen was the `Display output is not active.` message.

I came here looking for a solution. In the meantime, the login UI appeared. 🤷‍♂️ 

These are the settings I have, in case it helps.

![image](https://github.com/user-attachments/assets/e249d7ab-8b61-4369-b86f-1939f75c9404)

The machine running UTM is a MacBook Pro M1 Max.

**Reply by mokagio** [reply]
I spoke too soon...

After logging in, it went back to the error. I let it sit for a while, but unlike on boot, it did not move forward:
<img width="1392" alt="Screenshot 2024-09-24 at 5 09 46 PM" src="https://github.com/user-attachments/assets/3278b394-6c1c-48e0-8ec1-7525ec1956ef">

**Reply by mokagio** [reply]
Switching to `virtio-gpu-gl-pci (GPU Supported)` seems to do the trick for me

<img width="1573" alt="Screenshot 2024-09-24 at 7 30 24 PM" src="https://github.com/user-attachments/assets/2b3566b9-9a87-4281-b3d0-13a590c9edba">

<img width="900" alt="Screenshot 2024-09-24 at 7 33 03 PM" src="https://github.com/user-attachments/assets/3440e4b1-34c9-4045-9e79-f90934b3ceb5">

**Reply by MondeAnna** [reply]
Did the trick. Thanks

**Comment by xpitt** [comment]
my setup: utm 4.6.0, macbook pro M1 max 16" 2021, macos 15.2 beta 3, ubuntu 24.0.4. Tried all the settings change above, in the best arrangement (utm settings default, display virtio-rambf-gl (GPU supported) ) the `Display Output is not active' screen remains for about two seconds and a half. The same happens if I install ubuntu activating in utm the box for apple virtualization (experimental). Upon installing in Oakley's app Liviable I ca see the console output during startup; see attachment of a hard copy of the screen. After `Finished NetworkManager-wait-online.service....' the output stopped for about 20 secs, then for about two minutes after `Job systemd-network...', then for another 20 secs (no output recorded here) before finally the login window appeared. Maybe some linux wiz can find out what's wrong, I can't but suspect the culprit is not the utm display setting.

Any idea? Thanks anyway


<img width="749" alt="Screenshot 2024-11-17 at 12 09 07" src="https://github.com/user-attachments/assets/c2a93539-66c1-4180-8df6-4c20b6ba701a">

**Comment by xpitt** [comment]
Where I wrote for two seconds and a half I meant two minutes and a half, sorry

**Comment by jstoppa** [comment]
in my case, I was using a MacBook Pro with 2 monitors connected, when I move the UTM app to the main laptop screen then I stopped having this issue

**Comment by mickey-doyle** [comment]
Everyone has a different fix for this issue, so I figured I'd post mine in case it helps anyone. For me, what fixed this was deleting the VM and making a new one, except this time I left the storage space at the default 64gb. I had tried decreasing storage space to 32gb the first time around. Not positive this was what fixed the issue but it was the only thing I did differently and it did the trick.

**Reply by linearhover** [reply]
It did work for me, reinstalled vm with gpu acceleration, left ram and disk storage unchanged. Thanks for sharing!

**Comment by shaker6cat** [comment]
My configuration is here:
Macbook Air, M1 chip, Ventura 13.7 Os.
I use UTM to create a virtual machine with the ubuntu-24.04.1-live-server-arm64.iso. 

First of all, I open the UTM settings-display. Then select the **ANGLE(OpenGL)**.
<img width="597" alt="image" src="https://github.com/user-attachments/assets/2a7bfc89-de57-47bc-a7d0-af8b6210c16e">

Secondly, I open the virtual machine editing-display. Then I select the **virtio-gpu-gl-pic (GPU-Supported)**.
<img width="814" alt="image" src="https://github.com/user-attachments/assets/bd1c9b84-4aec-4aa8-a21a-bbe188ab5674">

Thirdly, I start the virtual machine and install the ubuntu-desktop with the aptitude tools. And then reboot.
You have to wait a few minutes for this screen to work. Maybe five or six minutes. **Just swipe your phone, laugh and keep waiting**.
<img width="713" alt="image" src="https://github.com/user-attachments/assets/b2f15ce4-3c2f-4cc8-b280-7c94fc01a4ea">

Here it is:
<img width="1254" alt="image" src="https://github.com/user-attachments/assets/50bd4224-7720-46c1-a955-e0c7e1689566">

Wish you success!

**Comment by devartes** [comment]
mine has this problem and when it allows the installation, it does not finish the installation when restarting, when it restarts it returns to the previous state to install

**Reply by YasithR** [reply]
Did it give any indication that the install was completed successfully? If so power off the VM and remove the iso from the Drives section

**Comment by 0xb0mb3r** [comment]
Yep adding a serial device is the fix for this. at least for the installation part. and dont forget to install spice tools once the installation is done
`$ sudo apt install qemu-guest-agent`
`$ sudo apt install spice-vdagent`

**Reply by David9006** [reply]
I wonder how long you waited for the installation to finish?

**Comment by Aiman-G** [comment]
In case someone still experiencing this problem: 

For me the "Display output is not active " will stay there for minutes, then the desktop will show. I found that the problem is not from UTM, it is from the Ubuntu Desktop, it causes a conflict due to conflicting network management services. I found a post that explains the conflict as :

 " Ubuntu uses one of two services to manage networking:

systemd-networkd: Default for servers.
NetworkManager: Default for desktops.
If both services (or the wrong one) are enabled, the boot process may timeout, causing delays. " 

 So I disabled the systemd-networkd serivece using this command " systemctl disable systemd-networkd.service " , now the screen still shows, but for a very short time. 

for more , here is the post helped solve this problem : [https://hadna.space/notes/38-ubuntu-desktop-arm](url)

**Reply by pasiphaegh** [reply]
Thanks @Aiman-G and @Noodlefishh! Your two solutions definitely solved my problem. Here is how I used them:

1. I added a new Serial device to the VM using the default Built-in Terminal.
2. I ran Ubuntu again and saw two windows: 1) main window with "Display output is not active", and 2) tty1 in a separate window, showing the full boot process. I could see the boot process becoming stuck on `systemd-networkd.service`. The terminal paused for a minute but then continued on without waiting for the network, which allowed me to logon to the terminal, while the main window still just showed "Display output is not active".
3. After logging into the terminal window, I issued the command @Aiman-G provided above, which was `systemctl disable systemd-networkd.service`, and I received a response that the service had been removed but its dependencies were still active. I assumed the warning was for this session only so I rebooted - see next step.
4. I rebooted with `sudo shutdown -r now`. The reboot was successful and the two windows appeared again. The main window showed "Display output is not active" for a few seconds only while the secondary terminal window showed that the boot process did not become stuck on the networking. The main window continued to the Ubuntu login screen after only a few seconds.

Thanks so much!

**Reply by tfgcampos** [reply]
Thank you so much!

This step should be added to official guide at https://docs.getutm.app/guides/ubuntu/.

**Comment by YasithR** [comment]
Linux version - Kali Linux 2024 arm64
Installing on - Macbook Air M1 
UTM - 4.6.3

I had the same problem with getting the Display Output is not active when selecting either  _virtio-gpu-gl-pic (GPU-Supported)_ or _virtio-ramfb-gl (GPU Supported)._ 

What I did was go into the Virtual Machine settings in UTM and **added a new device (Serial) and then powered on the VM**. Now it gave me a nice terminal window that asked me for the hostname,username and other settings. **Keep in mind that the bigger window still said Display Output is not active while the smaller terminal window was working.** 

Continue with the installation. **When rebooting, turn the VM off and remove the iso from the Drives**. Then power on and voila! you will be inside Kali Linux

**Reply by FortisLeo** [reply]
Thanks a lot bro it worked

**Comment by 0xb0mb3r** [comment]
Yeah this works. But i also noticed that GNOME runs like shite on UTM. Very very poor performance. Xfce4 (default) runs way better

**Comment by smijolovic** [comment]
Same issue with 4.6.4 on ubuntu 24.04 without VDM. 

Went back to 4.5.4 and works fine with virtio-ramfb. Hope this gets fixed.

**Comment by devmarcosvinicius** [comment]
I created a new serial, and it worked.
After the installation, I removed the serial, and it continues to work normally.

**Reply by devmarcosvinicius** [reply]
Btw, the service systemd-networkd-wait-online takes a long time to start the system. I used `sudo systemctl disable systemd-networkd-wait-online` to disable it. After that, it starts much faster.

**Comment by Banz01** [comment]
I just wait a few minutes and login to Linux occurs

**Comment by nanni61** [comment]
Often the problem presents itself because Ubuntu (But not only it) is very slow to leave, some services must be disabled.

Use: 'systemd-analyze blame' in the terminal to see the charge times of the various services at the start of the operating system

**Comment by arindam-giri** [comment]
Wait for 1 min or so , it will work. With `ANGLE (OpenGL)` option as backend and `virtio-ramfb-gl`

**Reply by douyixuan** [reply]
Yes, all you need to do is to wait. 😂

**Reply by Graphic-lab** [reply]
I know ... 3 or 4 minutes waiting here and then it was ok🎁💥🚀

**Comment by da-** [comment]
For me the right answer was adding a serial device, which told me I needed to enter the password I had set on the encrypted boot disk.
The last time I installed Ubuntu 22.04 (on an Intel mac) it automatically displayed that graphically, but *shrug* Good 'nuff!

Good thing I didn't sit and wait for it to finish on its own ;)

**Comment by smijolovic** [comment]
Any plans to address this on M3/M4 for future releases?

For now frozen on 4.5.4 as it's the only version that works but I want to emulate x86_64 VMs with newer versions of QEMU for a processor that is newer than 2009.

**Comment by SKSADIRUDDIN** [comment]
yeah it worked for me as well , just wait few minutes M2 Macbook Air

**Comment by tomaytotomato** [comment]
I am unable to get a virtualized Ubuntu (22.04, 24.02) working using an `M4 Pro` (48gb RAM), using the latest version of UTM 4.6.5

These are the things I've tried:

* Enabling permissions on the Mac Privacy and Security for UTM
* Switching Graphics to **virtio-ramfb-gl** or `virtio-gl-pci`
* Waiting 5 minutes to show the screen, still nothing

**Comment by KanzakiPapillon** [comment]
I changed the parameter as per above suggestions and my screen is stuck on an attempting repairs, let's be hopeful but I am using Windows so slightly different
<img width="1870" alt="image" src="https://github.com/user-attachments/assets/ee794f73-c6ba-4b27-bb3e-7a4ebb3f2f9a" />

**Comment by Weugene** [comment]
I had the same problem. I activated a terminal on each VM and  have noticed error message:
 **Job systemd-networkd-wait-online.service/start running (37s / no limit)**
This means the systemd-networkd-wait-online.service is waiting too long (in this case, indefinitely) for a network connection to be fully up, which can slow down or block booting. This is common when Ubuntu is running in a virtualized environment like UTM on macOS M1.

![image](https://github.com/user-attachments/assets/5d4b1976-1253-4c14-a634-59af47a4ccb7)

Since I use NetworkManager (as shown in your logs), I don’t need systemd-networkd when VM is started just write this lines in your terminal:
```bash
sudo systemctl disable systemd-networkd-wait-online.service
sudo systemctl mask systemd-networkd-wait-online.service
```

**Comment by davidmarble** [comment]
After pounding my head on the table for hours trying various settings on a MacBook Pro w/ M4 Pro, following this post worked for me: https://zenn.dev/tetra2000/articles/x86_linux_on_m1

CPU: IvyBridge
GPU: VGA

My use case is an older version of Ubuntu Server, so GPU acceleration is not important to me.

**Comment by zell-wang** [comment]
我Macnimi m4，安装kali，新增一个串口即可
![1749128982512](https://github.com/user-attachments/assets/3d611711-a4e8-4d35-97a9-7d1e3a48b60e)
![1749129042536](https://github.com/user-attachments/assets/e8eae028-a67b-44a1-bbf2-c696d7c48a6f)

**Comment by h0ngzhi** [comment]
To work around this, I:
	1.	Created a serial terminal in the VM configuration alongside the main display output.
	2.	Ensured the disk size was at least 8 GiB 
	3.	Booted the VM as normal.
	4.	Initially, the display stayed blank  but after a short wait, output began appearing in the serial console window.

**Comment by facorazza** [comment]
I had configured a passphrase to encrypt the disk image so the desktop would never load waiting for user's input. I had to enable a serial terminal to insert the passphrase.

**Reply by pascalrobert** [reply]
Same thing here. UTM 4.7.5 + Fedora Workstation 44.

**Comment by exceptionSPG** [comment]
For me, adding a Serial devices (just the default one) worked. You have to wait a bit, once the start up console is completed on serial output.
